src/appleseed/renderer/modeling/edf/edf.cpp

```diff
@@ -128,4 +132,62 @@ void* EDF::evaluate_inputs(
     return data;
 }
 
+float EDF::get_max_contribution_scalar(const Source* source) const
+{
+    assert(source);
+
+    if (!source || !source->is_uniform())
+        return numeric_limits<float>::max();
+
+    float value;
+
+    source->evaluate_uniform(value);
+
+    return value;
+}
+
+float EDF::get_max_contribution_spectrum(const Source* source) const
+{
+    assert(source);
+
+    if (!source || !source->is_uniform())
+        return numeric_limits<float>::max();
+
+    Spectrum spectrum;
+
+    source->evaluate_uniform(spectrum);
+
+    return max_value(spectrum);
+}
+
+float EDF::get_max_contribution(const Source* input) const
```

Comment:
This method is a bit weird since it only works for inputs of type spectrum. In fact I don't think you're using it. I'd suggest to remove it.